### DISCUSSION TITLE
Getting multiple items from 1Password?

### DISCUSSION THREAD

**Discussion by oschrenk** [twpayne/chezmoi#2079] [General]
Is there a way to get and loop through multiple items in 1Password? 

- `onepasswordDetailsFields` requires UUID
- `onepasswordDocument` requires UUID
- `onepasswordItemFields`  requires UUID

That means I need to create the item in 1Password, retrieve it's UUID, then add an entry to my template. Say I need to manage multiple entries with a similar structure, say

```
realm=Realm 1
host=host1
user=joe
password=foo

realm=Realm 2
host=host2
user=jane 
password=bar

...
```

I would prefer to retrieve items based on say tags `op item list --format json --tags chezmoi,ivy` and loop through those in a template. That saves me touch the template for each item. I see two ways, having a specialized onepassword function that I can pass tags or a maybe a template function that accepts a generic cli command (eg `op item list --format json --tags chezmoi,ivy | jq -r .[].id` to extract the UUIDs) . 

Maybe it's already possible and I am missing a function?

**Comment by twpayne** [comment]
You should be able to do this with the `output` template function. To start, it would something like:

```
{{ $data := output "op" "item" "list" "--format" "json" "--tags" "chezmoi,ivy" | fromJson }}
{{ range $data }}
realm = {{ .realm }}
host = {{ .host }}
user = {{ .user }}
password = {{ .password }}
{{ end }}
```

The exact code required will depend on the structure of the data returned by `op item list`.
